internal/provider/resource_cloud_service.go

```diff
@@ -8,6 +8,105 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+// used for all Hosted Clouds
+func resourceServicesHostedUpdate(ctx context.Context, d *schema.ResourceData, m interface{}, fn func(string, string) (*packetfabric.CloudServiceConnCreateResp, error)) diag.Diagnostics {
+	c := m.(*packetfabric.PFClient)
+	c.Ctx = ctx
+	var diags diag.Diagnostics
+	cloudCID, ok := d.GetOk("id")
+	if !ok {
+		return diag.Errorf("please provide a valid cloud service id")
+	}
+	if d.HasChange("aws_account_id") ||
```

Comment:
@rjouhann, I would suggest to break this function into small functions.

Reply:
Adding `checkUpdatableFieldsHostedCloud()` function